solutions/cf-1555B.mdx

```diff
@@ -143,66 +163,81 @@ for _ in range(int(input())):
 <JavaSection>
 
 ```java
-import java.util.*;
 import java.io.*;
+import java.util.*;
+
+public class TwoTables {
+
+	static InputReader r = new InputReader(System.in);
+	static PrintWriter pw = new PrintWriter(System.out);
+
+	public static void main(String[] args) {
+		int n = r.nextInt();
+
+		for (int i = 0; i < n; i++) {
+			int W = r.nextInt();
+			int H = r.nextInt();
+			int x1 = r.nextInt();
+			int y1 = r.nextInt();
+			int x2 = r.nextInt();
+			int y2 = r.nextInt();
+			int w = r.nextInt();
+			int h = r.nextInt();
+
+			int answer = Integer.MAX_VALUE;
 
-public class Main {
-	public static void main(String[] args) throws IOException {
-		Kattio io = new Kattio();
-
-		int tc = io.nextInt();
-		for(int t = 0; t < tc; t++){
-			int totalWidth = io.nextInt();
-			int totalHeight = io.nextInt();
-			int x1 = io.nextInt();
-			int y1 = io.nextInt(); 
-			int x2 = io.nextInt(); 
-			int y2 = io.nextInt(); 
-
-			int w1 = x2 - x1;
-			int h1 = y2 - y1;
-
-			int w2 = io.nextInt();
-			int h2 = io.nextInt();
-			
-			// Calculating min and max X and Y values needed to fit in their places
-			int leftPlace = w2;
-			int rightPlace = totalWidth - w2;
-			int topPlace = totalHeight - h2;
-			int bottomPlace = h2;
-
-			// Calculating distance needed to move first rectangle first rectangle
-			int amountNeededLeft = Math.max(leftPlace - x1, 0);
-			int amountNeededRight = Math.max(x2 - rightPlace, 0);
-			int amountNeededTop = Math.max(y2 - topPlace, 0);
-			int amountNeededBottom = Math.max(bottomPlace - y1, 0);
-
-			// Checking bounds of the rectangles
-			if (w1 + w2 > totalWidth) {
-				amountNeededLeft = Integer.MAX_VALUE;
-				amountNeededRight = Integer.MAX_VALUE;
+			if ((x2 - x1) + w <= W) { // Check if combined table widths can fit in the room
+				int left = Math.max(0, w - x1); // Determine overlap when placing the table on the left
+				int right = Math.max(0, x2 - (W - w)); // Determine overlap when placing the table on the
+				// right
+				answer = Math.min(left, right);
 			}
-			
-			if (h1 + h2 > totalHeight) {
-				amountNeededTop = Integer.MAX_VALUE;
-				amountNeededBottom = Integer.MAX_VALUE;
+			if ((y2 - y1) + h <= H) { // Check if combined table heights can fit in the room
+				int top = Math.max(0, y2 - (H - h));
+				int bottom = Math.max(0, h - y1);
+				int minTB = Math.min(top, bottom);
+				answer = Math.min(answer, minTB);
 			}
 
-			int ans = Integer.MAX_VALUE;
-			if (ans > amountNeededLeft) ans = amountNeededLeft;
-			if (ans > amountNeededRight) ans = amountNeededRight;
-			if (ans > amountNeededTop) ans = amountNeededTop;
-			if (ans > amountNeededBottom) ans = amountNeededBottom;
+			pw.println(answer == Integer.MAX_VALUE ? -1 : answer);
+		}
+		pw.close();
+	}
 
+	static class InputReader {
```

Comment:
i'd just use `BufferedReader` or `Kattio`

Reply:
Hi there, this is a BufferedReader based I/O template I copied from Darren Yao's Intro to USACO. I would like to keep it if possible to save on refactoring. I note that personal I/O templates are allowed. I have, however, now collapsed it with a CodeSnip.